File: Chapter09/HelpDeskVB/HelpDeskVB/HelpDeskVB.DesktopClient/Screens/CreateNewIssueResponse.lsml.vb
'You can use and redistribute the code from this book (and sample application) for non-commercial and 
'most commercial purposes as long as you acknowledge the source and authorship. 
'You should note the source of the code in any documentation as well as in the program code itself (as a comment). 
'The acknowledgment should include author, title, publisher, and ISBN. 
'An example of such an acknowledgment would be "Derived from Listing 2-10, LightSwitch 2015 by Tim Leung, published by Apress, ISBN 978-1-4842-0767-3".

Namespace LightSwitchApplication

    Public Class CreateNewIssueResponse
        'Listing 9-14. Handling the TextBox KeyUp event 

        Private Sub CreateNewIssueResponse_InitializeDataWorkspace(
   saveChangesTo As List(Of Microsoft.LightSwitch.IDataService))
            Me.IssueResponseProperty = New IssueResponse()

            Dim control = Me.FindControl("ResponseText")
            AddHandler control.ControlAvailable,
        AddressOf TextBoxAvailable

            ResponseTextCount = 1000

        End Sub

        Private Sub TextBoxAvailable(
   sender As Object, e As ControlAvailableEventArgs)
            AddHandler CType(e.Control,
        System.Windows.Controls.TextBox).KeyUp,
            AddressOf TextBoxKeyUp
        End Sub

        Private Sub TextBoxKeyUp(
    sender As Object, e As System.Windows.RoutedEventArgs)

            Dim textbox = CType(sender, System.Windows.Controls.TextBox)
            ResponseTextCount = 1000 - textbox.Text.Count()
        End Sub




        Private Sub CreateNewIssueResponse_Saved()
            ' Write your code here.
            Me.Close(False)
            Application.Current.ShowDefaultScreen(Me.IssueResponseProperty)
        End Sub

    End Class

End Namespace

File: Chapter09/HelpDeskCS/HelpDeskCS/HelpDeskCS.DesktopClient/Screens/CreateNewIssueResponse.lsml.cs
// You can use and redistribute the code from this book (and sample application) for non-commercial and 
// most commercial purposes as long as you acknowledge the source and authorship. 
// You should note the source of the code in any documentation as well as in the program code itself (as a comment). 
// The acknowledgment should include author, title, publisher, and ISBN. 
// An example of such an acknowledgment would be "Derived from Listing 2-10, LightSwitch 2015 by Tim Leung, published by Apress, ISBN 978-1-4842-0767-3".

using Microsoft.LightSwitch.Presentation.Extensions;
using Microsoft.LightSwitch.Presentation;
using Microsoft.LightSwitch.Framework.Client;
using Microsoft.LightSwitch;
using System.Collections.Generic;
using System.IO.IsolatedStorage;
using System.IO;
using System.Linq;
using System;

namespace LightSwitchApplication
{
    public partial class CreateNewIssueResponse
    {
        //Listing 9-14. Handling the TextBox KeyUp event
        partial void CreateNewIssueResponse_InitializeDataWorkspace(global::System.Collections.Generic.List<global::Microsoft.LightSwitch.IDataService> saveChangesTo)
        {
            this.IssueResponseProperty = new IssueResponse();
            this.FindControl("ResponseText").ControlAvailable += TextBoxAvailable;
            ResponseTextCount = 1000;
        }

        private void TextBoxAvailable(object sender, ControlAvailableEventArgs e)
        {
            ((System.Windows.Controls.TextBox)e.Control).KeyUp += TextBoxKeyUp;
        }

        private void TextBoxKeyUp(object sender, System.Windows.RoutedEventArgs e)
        {
            var textbox = (System.Windows.Controls.TextBox)sender;
            ResponseTextCount = 1000 - textbox.Text.Count();
        }

        partial void CreateNewIssueResponse_Saved()
        {
            // Write your code here.
            this.Close(false);
            Application.Current.ShowDefaultScreen(this.IssueResponseProperty);
        }



    }
}